store/heightsub.go

```diff
@@ -35,7 +36,15 @@ func (hs *heightSub[H]) Height() uint64 {
 
 // SetHeight sets the new head height for heightSub.
 func (hs *heightSub[H]) SetHeight(height uint64) {
-	hs.height.Store(height)
+	for {
+		curr := hs.height.Load()
+		if curr >= height {
+			return
+		}
+		if hs.height.CompareAndSwap(curr, height) {
+			return
+		}
+	}
```

Comment:
This is correct when there are multiple routines writing into the atomic, however, we have only one, so this seems redundant